packages/flutter_tools/templates/plugin/android-java.tmpl/src/main/java/androidIdentifier/pluginClass.java.tmpl

```diff
@@ -21,6 +22,20 @@ public class {{pluginClass}} implements FlutterPlugin, MethodCallHandler {
     channel.setMethodCallHandler(new {{pluginClass}}());
   }
 
+  // This function is optional and equivalent to onAttachedToEngine. It supports the old
+  // pre-Flutter-1.12 Android projects. Plugin developers are encouraged to continue supporting
+  // plugin registration via this function while apps migrate to use the new Android APIs
+  // post-flutter-1.12.
+  //
+  // Developers are encouraged to share logic between onAttachedToEngine and registerWith to keep
+  // them functionaly equivalent. Only one of onAttachedToEngine or registerWith will be called
+  // depending on the user's project. onAttachedToEngine or registerWith must both be defined
+  // in the same class.
```

Comment:
@sfshaza2 do you prefer to use third-person or second-person when referring to developers in docs?

Reply:
Generally I think we prefer second person. https://developers.google.com/style/person

Reply:
https://github.com/flutter/website/pull/3007#discussion_r325219526 oh you're right. Shams actually pointed this out to me before. Thanks